### DISCUSSION TITLE
Wildcard Subdomains for Multi-tenancy

### DISCUSSION THREAD

**Discussion by ZebTheWizard** [coollabsio/coolify#1905] [Q&A] [answered]
Hello, I have a multi-tenant PHP application where each user has a unique subdomain `<username>.example.com`. Looks like it's a little bit more involved than configuring an NGINX file. What needs to be configured to make this work?

**Comment by ZebTheWizard** [comment, marked as answer]
Update: for wildcard subdomains I had to do something like this.
```
traefik.http.routers.http-0-zwos0g8.rule=HostRegexp(`{subdomain:([a-zA-Z]+\.)?}example.com`) && PathPrefix(`/`)

traefik.http.routers.https-0-zwos0g8.rule=HostRegexp(`{subdomain:([a-zA-Z]+\.)?}example.com`) && PathPrefix(`/`)
```
 
Last thing I need to figure out is how to ensure Traefik, Caddy, Nginx, etc. permanently redirect to HTTPS while keeping all the headers. Currently seems like load balancing is not setting x-forwarded-proto correctly among other things. Anyone know?

**Reply by jrvidotti** [reply]
I've managed to make it work with fully escaped regexes and no "{subdomain:...}" notation:

```
HostRegexp(`([a-zA-Z0-9-]+\.)?example\.com`)
```

**Comment by ZebTheWizard** [comment]
Turns out when you configure a DNS challenge you should double check the provider you're generating an API key for is your nameserver and not your VPS. 🤦‍♂️ 

Setting Cloudflare SSL mode to "Full (strict)" helped me figure this out. Kept getting 526 errors.

**Comment by peaklabs-dev** [comment]
@ZebTheWizard is this resolved?

**Comment by tiagomichaelsousa** [comment]
I had the same issue today and created a thread in discord under coolify support.
@Cinzya actually helped solved the issue and the problem was actually mine due do a `'` 😂 developers nightmares.

Either way, while doing this, I suggested/gave feedback of adding the possiblity to configure your domains directly in coolify UI using * for wildcard subdomains.

As an example:
<img width="717" height="103" alt="image" src="https://github.com/user-attachments/assets/37a7d4d8-76c5-4385-91e7-321a09f31705" />

With this configuration, coolify could automatically add the correct container labels and support wildcard subdomains for the application out of the box.

Thoughts? 🤔

**Reply by ldenblyd** [reply]
this feature would be awesome !

**Reply by marcelovicentegc** [reply]
It occurs to me that this is what @peaklabs-dev and @DeVoresyah proposed on #2391.

Would be awesome to have this feature so easily configurable!
